go/libraries/doltcore/branch_control/branch_control.go

```diff
@@ -56,22 +66,86 @@ func init() {
 
 // CreateController returns an empty *Controller.
 func CreateController(ctx context.Context) *Controller {
-	accessTbl := newAccess("", "")
-	return &Controller{
-		Access:    accessTbl,
-		Namespace: newNamespace(accessTbl, "", ""),
-	}
+	return CreateControllerWithSuperUser(ctx, "", "")
 }
 
 // CreateControllerWithSuperUser returns a controller with the given user and host set as an immutable super user.
 func CreateControllerWithSuperUser(ctx context.Context, superUser string, superHost string) *Controller {
+	//TODO: put in the context
 	accessTbl := newAccess(superUser, superHost)
 	return &Controller{
 		Access:    accessTbl,
 		Namespace: newNamespace(accessTbl, superUser, superHost),
 	}
 }
 
+// LoadData loads the data from the given location into the controller.
+func LoadData(ctx context.Context, branchControlFilePath string, doltConfigDirPath string) error {
+	//TODO: load into the context's controller
+	if !enabled {
+		return nil
+	}
+
+	// Do not attempt to load from an empty file path
+	if len(branchControlFilePath) == 0 {
+		return nil
+	}
+
+	StaticController.branchControlFilePath = branchControlFilePath
+	StaticController.doltConfigDirPath = doltConfigDirPath
+	data, err := os.ReadFile(branchControlFilePath)
+	if err != nil && !goerrors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	// Nothing to load so we can return
+	if len(data) == 0 {
+		return nil
+	}
+	position := uint64(0)
+	// The Deserialize functions acquire write locks, so we don't acquire them here
+	if err = StaticController.Access.Deserialize(data, &position); err != nil {
+		return err
+	}
+	if err = StaticController.Namespace.Deserialize(data, &position); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SaveData saves the data from the context's controller to the location pointed by it.
+func SaveData(ctx context.Context) error {
```

Comment:
Probably makes more sense as `Serialize(ctx, filepath)`, but maybe that can wait until it's a method on the controller object?